app/celery/broadcast_message_tasks.py

```diff
@@ -50,35 +133,56 @@ def send_broadcast_provider_message(broadcast_event_id, provider):
 
     cbc_proxy_provider_client = cbc_proxy_client.get_proxy(provider)
 
-    if broadcast_event.message_type == BroadcastEventMessageType.ALERT:
-        cbc_proxy_provider_client.create_and_send_broadcast(
-            identifier=str(broadcast_provider_message.id),
-            message_number=formatted_message_number,
-            headline="GOV.UK Notify Broadcast",
-            description=broadcast_event.transmitted_content['body'],
-            areas=areas,
-            sent=broadcast_event.sent_at_as_cap_datetime_string,
-            expires=broadcast_event.transmitted_finishes_at_as_cap_datetime_string,
+    try:
+        if broadcast_event.message_type == BroadcastEventMessageType.ALERT:
+            cbc_proxy_provider_client.create_and_send_broadcast(
+                identifier=str(broadcast_provider_message.id),
+                message_number=formatted_message_number,
+                headline="GOV.UK Notify Broadcast",
+                description=broadcast_event.transmitted_content['body'],
+                areas=areas,
+                sent=broadcast_event.sent_at_as_cap_datetime_string,
+                expires=broadcast_event.transmitted_finishes_at_as_cap_datetime_string,
+            )
+        elif broadcast_event.message_type == BroadcastEventMessageType.UPDATE:
+            cbc_proxy_provider_client.update_and_send_broadcast(
+                identifier=str(broadcast_provider_message.id),
+                message_number=formatted_message_number,
+                headline="GOV.UK Notify Broadcast",
+                description=broadcast_event.transmitted_content['body'],
+                areas=areas,
+                previous_provider_messages=broadcast_event.get_earlier_provider_messages(provider),
+                sent=broadcast_event.sent_at_as_cap_datetime_string,
+                expires=broadcast_event.transmitted_finishes_at_as_cap_datetime_string,
+            )
+        elif broadcast_event.message_type == BroadcastEventMessageType.CANCEL:
+            cbc_proxy_provider_client.cancel_broadcast(
+                identifier=str(broadcast_provider_message.id),
+                message_number=formatted_message_number,
+                previous_provider_messages=broadcast_event.get_earlier_provider_messages(provider),
+                sent=broadcast_event.sent_at_as_cap_datetime_string,
+            )
+    except CBCProxyRetryableException as exc:
```

Comment:
I'm still wondering if we catch any exception that isn't a `CBCProxyFatalException` if we should retry? Just wondering if there any other types of flakes that we might get that wouldn't trigger a `CBCProxyRetryableException` that I'd rather retry unless we are absolutely confident they shouldn't. Like maybe some other random network flake when connecting to AWS or something